tests/integ/modin/series/test_items.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import numpy as np
+import pandas as native_pd
+import pytest
+
+from tests.integ.modin.utils import create_test_series, eval_snowpark_pandas_result
+from tests.integ.utils.sql_counter import SqlCounter, sql_count_checker
+
+
+def assert_items_results_equal(snow_result, pandas_result) -> None:
+    snow_list = list(snow_result)
+    pandas_list = list(pandas_result)
+    assert len(snow_list) == len(pandas_list), "lengths of items are not equal."
+    if len(snow_list) == 0:
+        # Expect no queries if there are no columns.
+        with SqlCounter(query_count=0):
+            return
+    for ((snow_index, snow_value), (pandas_index, pandas_value)) in zip(
+        snow_list, pandas_list
+    ):
+        with SqlCounter(query_count=0):
+            assert snow_index == pandas_index
+            # Workaround using NumPy since nan != nan
```

Comment:
This seems  a little awkward. Can we use [`assert_values_equal`](https://github.com/snowflakedb/snowpark-python/blob/560d56318208d9285c381897a46f0e747755820c/tests/integ/modin/utils.py#L481)?

Reply:
Oh didn't realize that util function existed, neat!